src/main/java/org/springframework/cloud/deployer/spi/scheduler/cloudfoundry/CloudFoundryAppScheduler.java

```diff
@@ -220,8 +257,14 @@ private void scheduleTask(String appName, String scheduleName,
 	private String stageTask(ScheduleRequest scheduleRequest) {
 		logger.debug(String.format("Staging Task: ",
 				scheduleRequest.getDefinition().getName()));
+		Map<String, String> properties = new HashMap<>(scheduleRequest.getDefinition().getProperties());
+		properties.put(SCHEDULER_TASK_DEF_NAME_KEY,
+				scheduleRequest.getDefinition().getName());
```

Comment:
This will be same as that of the schedule name and hence, not sure why we need to store it here. or, did you mean to store the the actual task application name that will be launched using this dataflow task launcher?

Reply:
Yes.    The name of the droplet that is deployed to cloud foundry will have the name of the schedule (for example my-schedule).   This property will store the name of the task definition (my-task-definition) that is associated with droplet.